File: VB/NavBarAndPrism.ModuleB/ModuleB.vb
Imports System
Imports System.Net
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Documents
Imports System.Windows.Ink
Imports System.Windows.Input
Imports System.Windows.Media
Imports System.Windows.Media.Animation
Imports System.Windows.Shapes
Imports NavBarAndPrism.Views
Imports System.ComponentModel.Composition
Imports Prism.Mef.Modularity
Imports Prism.Modularity
Imports Prism.Regions

Namespace NavBarAndPrism.Modules
	<ModuleExport(GetType(ModuleB))>
	Public Class ModuleB
		Implements IModule

		<Import>
		Public RegionManager As IRegionManager
        Public Sub Initialize() Implements IModule.Initialize
            RegionManager.RegisterViewWithRegion("NavBarControlRegion", GetType(ModuleBView))
        End Sub
    End Class
End Namespace

File: CS/NavBarAndPrism.ModuleB/ModuleB.cs
using System;
using System.Net;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Documents;
using System.Windows.Ink;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Animation;
using System.Windows.Shapes;
using Microsoft.Practices.Prism.Modularity;
using Microsoft.Practices.Prism.Regions;
using NavBarAndPrism.Views;
using Microsoft.Practices.Prism.MefExtensions.Modularity;
using System.ComponentModel.Composition;

namespace NavBarAndPrism.Modules {
    [ModuleExport(typeof(ModuleB))]
    public class ModuleB : IModule {
        [Import]
        public IRegionManager RegionManager;
        public void Initialize() {
            RegionManager.RegisterViewWithRegion("NavBarControlRegion", typeof(ModuleBView));
        }
    }
}
